huaweicloud/services/dli/resource_huaweicloud_dli_permission.go

```diff
@@ -0,0 +1,350 @@
+package dli
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/chnsz/golangsdk"
+	"github.com/chnsz/golangsdk/openstack/dli/v1/auth"
+	"github.com/chnsz/golangsdk/openstack/dli/v1/flinkjob"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
+)
+
+func ResourceDliPermission() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceDliPermissionCreate,
+		ReadContext:   resourceDliPermissionRead,
+		DeleteContext: resourceDliPermissionDelete,
+		UpdateContext: resourceDliPermissionUpdate,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"user_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"object": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"privileges": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"is_admin": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceDliPermissionCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	client, err := config.DliV1Client(config.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("Error creating HuaweiCloud DLI v1 client: %s", err)
+	}
+
+	obj := d.Get("object").(string)
+	userName := d.Get("user_name").(string)
+
+	if strings.HasPrefix(obj, "queues.") {
+		queueName := strings.Replace(obj, "queues.", "", 1)
+		opts := auth.GrantQueuePermissionOpts{
+			QueueName: queueName,
+			UserName:  userName,
+			Action:    "update",
+		}
+
+		if privileges, ok := d.GetOk("privileges"); ok {
+			for _, v := range privileges.([]interface{}) {
+				opts.Privileges = append(opts.Privileges, v.(string))
+			}
+		}
+
+		rst, createErr := auth.GrantQueuePermission(client, opts)
+		if createErr != nil {
+			return fmtp.DiagErrorf("Error grant permission in DLI: %s", createErr)
+		}
+
+		if rst != nil && !rst.IsSuccess {
```

Comment:
When re-authorizing or modifying resources, is it the same as the first authorization?

Reply:
all authorizate is update permissions